charts/agent/templates/_helpers.tpl

```diff
@@ -537,25 +537,83 @@ true
 {{- end }}
 {{- end }}
 
+{{/* Check if semver. The regex is from the code of the library Helm uses for semver. */}}
+{{- define "agent.isSemVer" -}}
+    {{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" . }}
+        true
+    {{- end -}}
+{{- end -}}
+
 {{/* Return the name of the local forwarder configmap */}}
 {{- define "agent.localForwarderConfigMapName" }}
 {{- include "agent.configmapName" . | trunc 46 | trimSuffix "-" | printf "%s-local-forwarder" }}
 {{- end }}
 
 {{- define "agent.enableHttpProbes" }}
 {{- if not (include "agent.gke.autopilot" .) }}
-{{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" .Values.image.tag }}
-{{- if semverCompare ">= 12.18.0-0" .Values.image.tag }}
+{{- if and (include "agent.isSemVer" .Values.image.tag) (semverCompare ">= 12.18.0-0" .Values.image.tag) }}
 {{- printf "true" -}}
 {{- end }}
 {{- end }}
 {{- end }}
-{{- end }}
 
 {{- define "agent.enableFalcoBaselineSecureLight" }}
-{{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" .Values.image.tag }}
-{{- if semverCompare ">= 12.19.0-0" .Values.image.tag }}
+{{- if and (include "agent.isSemVer" .Values.image.tag) (semverCompare ">= 12.19.0-0" .Values.image.tag) }}
 {{- printf "true" -}}
 {{- end }}
 {{- end }}
+
+{{- define "agent.annotationsSection" }}
+  {{- if (include "agent.gke.autopilot" .) }}
+    {{- printf "annotations:" }}
+    {{- printf "autopilot.gke.io/no-connect: \"true\"" | nindent 2 }}
```

Comment:
We can avoid to use printf here as it make it more readable:
```suggestion
annotations:
  autopilot.gke.io/no-connect: "true"
```

Reply:
IMHO that's debatable, I find it more readable with printf, which also allows me to be consistent with the line where I have to use `toYaml`

Reply:
FYI is the same approach we use in all the charts here and is also the [default behaviour from helm](https://github.com/helm/helm/blob/main/pkg/chartutil/create.go#L259-L262) (so is a well known pattern in the ecosystem), honestly I'll like to keep consistency here. 🙏